admin_manual/configuration/server/occ_command.rst

```diff
@@ -1893,8 +1896,50 @@ To delete a user, you use the ``user:delete`` command, as in the example below:
 ::
 
  sudo -u www-data php occ user:delete fred
-   
-   
+
+Expiring a User’s Password
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To expire a user’s password at a given time, use the ``user:expire-password`` command, as in the example below:
+
+::
+
+  # Expire the user "frank"’s password 24 hours prior
+  occ user:expire-password frank
+
+  # Expire the user "frank"’s password in 2 days
+  occ user:expire-password frank '+2 days'
+
+  # Expire the user "frank"’s password on the 15th of August 2005, at 15:52:01 in the local timezone
+  occ user:expire-password frank '2005-08-15T15:52:01+00:00'
+
+  # Expire the user "frank"’s password on the 15th of August 2005, at 15:52:01 UTC
+  occ user:expire-password frank '15-Aug-05 15:52:01 UTC'
+
+
+The command accepts expiry dates in any of `PHP’s supported date and time formats`_, and defaults to the previous day if an expiry date is not supplied.
+When the command runs, you will see a confirmation of when the password is set to expire, printed to the console, as in the example output below.
+
+::
+
+  The password for frank is set to expire on 2018-07-12 13:15:28 UTC.
+
+Please Be Aware That
+````````````````````
+
+- Enabling "days until user password expires" doesn't automatically update password expiration dates, for existing users. Administrators will need to run the ``occ user:expire-password`` command, with a specific value to force expiration either directly or after a few days
+- When changing the value of "days until user password expires", expiration dates of the "forced" users would stay the same and need updating by running the occ command again.
```

Comment:
After seeing the way the code works, the "forced expiration dates" are actually stored as "forced last password change dates". e.g. if "days until user password expires" = 90 and I force expire ``user1`` password to expire 30 days in the future, then a "forced" password history entry is put 60 days in the past. When we get 30 days in the future, that password history entry will be 90 days old, and so the password will expire.

If, before that 30 days in the future has come, the admin changes "days until user password expires" from 90 to 180, then the "forced" expiry will happen another 90 days later than initially expected. But all other "ordinary" password expiry will also happen 90 days later.

How to say the effect of the above in a few words in this doc?
@settermjd

Reply:
at this point I'm not sure whether those technical details are worth mentioning.

what an admin must know:
- after enabling password expiration policy, need to run the occ command to initiate expiry for existing users
- if changing password expiration rule days (ex: changing from 90 to 30 days), also need to run occ to initiate said expiry for the users

failing to run occ will either result in users password not expiring at all in the former case, or expiring based on the old rule

Reply:
Thanks for the further clarification as, given the information provided, it's not been easy. That last question's a good one too.

Reply:
@phil-davis, am I correct in the following understanding?

1. When the `occ user:expire` command is run and a time period (+40 days, +2 weeks, etc) is provided for the expiry date, and a password policy _is already in place_, the forced expiry date created will be adjusted based on the active password policy?
2. If the active password policy is changed after the `occ user:expire` command is run, then the expiry date of users whose passwords have already been forcibly expired, _after the policy was enabled_, will be updated to reflect the policy change?

---

I'm going to do some further digging to see what happen when a password policy is in place but a specific date is provided.

Reply:
1 whenever occ user:expire-password is run, the internal stuff is done so that, based on the current expiry days policy, the password will expire on the date specified.
2 when the expiry days policy is changed, the effective expiry date of passwords for all users slides in or out by the amount of the change.
This includes users that have had their password expiry date specified by the admin using the occ command.